src/posit/connect/resources.py

```diff
@@ -43,3 +46,254 @@ def update(self, *args, **kwargs):
+        response as JSON, and instantiates each item using `cls`.
+
+    __getitem__(index) -> Union[T_co, Sequence[T_co]]
+        Retrieve an item or slice from the sequence. Indexing follows the standard Python sequence semantics.
+
+    __len__() -> int
+        Return the number of items in the sequence.
+
+    __str__() -> str
+        Return a string representation of the cached data.
+
+    __repr__() -> str
+        Return a detailed string representation of the cached data.
+
+    reload() -> ActiveSequence
+        Clear the cache and mark to reload the data from the API on the next operation.
+    """
+
+    def __init__(self, cls: Type[T_co], ctx: Context, parent: Optional[Active] = None):
+        super().__init__()
+        self._cls = cls
+        self._ctx = ctx
+        self._parent = parent
+        self._cache = None
+
+    @property
+    @abstractmethod
+    def _endpoint(self) -> str:
+        """
+        Abstract property to define the endpoint URL for the GET request.
+
+        Subclasses must implement this property to return the API endpoint URL that will
+        be queried to fetch the data.
+
+        Returns
+        -------
+        str
+            The API endpoint URL.
+        """
+        raise NotImplementedError()
+
+    @property
+    def _data(self) -> List[T_co]:
+        """
+        Fetch and cache the data from the API.
+
+        This method sends a GET request to the `_endpoint` and parses the response as a list of JSON objects.
+        Each JSON object is used to instantiate an item of type `T_co` using the class specified by `_cls`.
+        The results are cached after the first request and reused for subsequent access unless reloaded.
+
+        Returns
+        -------
+        List[T_co]
+            A list of items of type `T_co` representing the fetched data.
+        """
+        if self._cache:
+            return self._cache
+
+        response = self._ctx.session.get(self._endpoint)
+        results = response.json()
+        self._cache = [self._cls(self._ctx, self._parent, **result) for result in results]
+        return self._cache
+
+    @overload
+    def __getitem__(self, index: int) -> T_co: ...
+
+    @overload
+    def __getitem__(self, index: slice) -> Sequence[T_co]: ...
+
+    def __getitem__(self, index):
+        return self._data[index]
+
+    def __len__(self) -> int:
+        return len(self._data)
+
+    def __str__(self) -> str:
+        return str(self._data)
+
+    def __repr__(self) -> str:
+        return repr(self._data)
+
+    def reload(self) -> "ActiveSequence":
+        """
+        Clear the cache and reload the data from the API on the next access.
+
+        Returns
+        -------
+        ActiveSequence
+            The current instance with cleared cache, ready to reload data on next access.
+        """
+        self._cache = None
+        return self
+
+
+class ActiveFinderMethods(ActiveSequence[T_co], ABC, Generic[T_co]):
```

Comment:
I believe this can be dropped as `ActiveSequence` covers it. (untested)

```suggestion
class ActiveFinderMethods(ActiveSequence[T_co], ABC):
```